### PR TITLE
#357 PeriodicBatchingSink now flushes on unhandled exceptions

### PR DESCRIPTION
**Pull request by chuwik**
Context for this in issue #357 

Not very happy with the name "ComeFromInvalidAppDomainContext" for the method, but couldn't think of a better one!

Also tried to create a unit test, but it's to do so when the AppDomain and unhandled exceptions are at play. Any suggestions for how to do this?

### CODE REVIEW

**Comment by nblumhardt** [comment]
Looks good. I think rather than give a name to that method, just inlining it would be as clear or clearer - how does that sound? Cheers!

**Comment by chuwik** [comment]
Sounds good, inlined it and fixed the condition: now it does "if it is of type UnhandledExceptionEventArgs and it's not terminating => don't flush"; the first part of that conditional was wrong before and would trigger the return for EventArgs coming from other places.

File: src/Serilog.FullNetFx/Sinks/PeriodicBatching/PeriodicBatchingSink.cs [hunk 1]
```diff
@@ -79,6 +84,7 @@ void CloseAndFlush()
 
             AppDomain.CurrentDomain.DomainUnload -= OnAppDomainUnloading;
             AppDomain.CurrentDomain.ProcessExit -= OnAppDomainUnloading;
+            AppDomain.CurrentDomain.UnhandledException += OnAppDomainUnloading;
```
Comment (nblumhardt) [inline src/Serilog.FullNetFx/Sinks/PeriodicBatching/PeriodicBatchingSink.cs]: Should be `-=`, not `+=` here

**Comment by nblumhardt** [comment]
Great, looking good! Just one line comment noted above. There's also a corresponding .NET 4.0 implementation in _Serilog-net40.sln_ that needs to be in sync. Nothing else from me! :-)

File: src/Serilog.FullNetFx/Sinks/PeriodicBatching/PeriodicBatchingSink.cs (hunk 1 above)
Comment (chuwik) [inline src/Serilog.FullNetFx/Sinks/PeriodicBatching/PeriodicBatchingSink.cs]: Right! Silly me!

**Comment by chuwik** [comment]
All fixed now! Thanks!

**Pull request merged**